controllers/firewall_controller.go

```diff
@@ -128,6 +136,109 @@ func (r *FirewallReconciler) validateFirewall(ctx context.Context, f firewallv1.
 	return nil
 }
 
+// migrateToClusterwideNetworkPolicy migrates old network policy objects to the new kind ClusterwideNetworkPolicy
+func (r *FirewallReconciler) migrateToClusterwideNetworkPolicy(ctx context.Context, f firewallv1.Firewall, log logr.Logger) error {
+	npsToIgnore := []string{"egress-allow-http", "egress-allow-https", "egress-allow-any", "egress-allow-dns", "egress-allow-ntp"}
+
+	var nps networking.NetworkPolicyList
+	if err := r.Client.List(ctx, &nps); err != nil {
+		return err
+	}
+
+	n := 0
+	for _, np := range nps.Items {
+		s := np.Spec
+		if len(s.PodSelector.MatchExpressions) != 0 || len(s.PodSelector.MatchLabels) != 0 {
+			continue
+		}
+
+		// is one of the old network policy objects like egress-allow-http that are replaced by cluster wide ones installed by gepm
+		if contains(npsToIgnore, np.Name) {
+			continue
+		}
+
+		cwnp, err := convert(np)
+		if err != nil {
+			return fmt.Errorf("could not migrate network policy to a cluster-wide np: %w", err)
+		}
+
+		if cwnp == nil {
+			// nothing to do here because network policy translates to an empty cwnp
+			continue
+		}
+
+		var current firewallv1.ClusterwideNetworkPolicy
+		err = r.Get(ctx, types.NamespacedName{Name: cwnp.Name, Namespace: firewallNamespace}, &current)
+
+		// cwnp already exists: don't try to merge or update - just ignore
+		if err == nil {
+			continue
+		}
+
+		if errors.IsNotFound(err) {
+			err = r.Client.Create(ctx, cwnp)
```

Comment:
This requires more RBAC rules for the firewall-controller, just to note and not to forget in the deployment

Reply:
see https://github.com/metal-stack/gardener-extension-provider-metal/pull/67/files